modules/local/orange/main.nf

```diff
@@ -0,0 +1,164 @@
+
+    input:
+    tuple val(meta), path(bam_metrics_somatic), path(bam_metrics_germline), path(flagstat_somatic), path(flagstat_germline), path(sage_somatic_dir), path(sage_germline_dir), path(smlv_somatic_vcf), path(smlv_germline_vcf), path(purple_dir), path(linx_somatic_anno_dir), path(linx_somatic_plot_dir), path(linx_germline_anno_dir), path(virusinterpreter_dir), path(chord_dir), path(sigs_dir), path(lilac_dir), path(cuppa_dir), path(isofox_dir)
+    val genome_ver
+    path disease_ontology
+    path cohort_mapping
+    path cohort_percentiles
+    path known_fusion_data
+    path driver_gene_panel
+    path ensembl_data_resources
+    path isofox_alt_sj
+    path isofox_gene_distribution
+    val pipeline_version
+
+    output:
+    tuple val(meta), path('output/*.orange.pdf') , emit: pdf
+    tuple val(meta), path('output/*.orange.json'), emit: json
+    path 'versions.yml'                          , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+
+    def pipeline_version_str = pipeline_version ?: 'not specified'
+
+    def virus_dir_arg = virusinterpreter_dir ? "-virus_dir ${virusinterpreter_dir}" : ''
+    def chord_dir_arg = chord_dir ? "-chord_dir ${chord_dir}" : ''
+    def sigs_dir_arg = sigs_dir ? "-sigs_dir ${sigs_dir}" : ''
+    def cuppa_dir_arg = cuppa_dir ? "-cuppa_dir ${cuppa_dir}" : ''
+    def plot_dir = linx_somatic_plot_dir.resolve('reportable/').toUriString().replaceAll('/$', '')
+
+    def normal_id_arg = meta.containsKey('normal_dna_id') ? "-reference_sample_id ${meta.normal_dna_id}" : ''
+    def normal_metrics_arg = bam_metrics_germline ? "-ref_sample_wgs_metrics_file ${bam_metrics_germline}" : ''
+    def normal_flagstat_arg = flagstat_germline ? "-ref_sample_flagstat_file ${flagstat_germline}" : ''
+    def normal_sage_dir = sage_germline_dir ? "-sage_germline_dir ${sage_germline_dir}" : ''
+    def normal_linx_arg = linx_germline_anno_dir ? "-linx_germline_dir ${linx_germline_anno_dir}" : ''
+
+    def rna_id_arg = meta.containsKey('tumor_rna_id') ? "-rna_sample_id ${meta.tumor_rna_id}" : ''
+    def isofox_dir_arg = isofox_dir ? '-isofox_dir isofox_dir__prepared/' : ''
+
+    def isofox_gene_distribution_arg = isofox_gene_distribution ? "-isofox_gene_distribution ${isofox_gene_distribution}" : ''
+    def isofox_alt_sj_arg = isofox_alt_sj ? "-isofox_alt_sj_cohort ${isofox_alt_sj}" : ''
+
+    """
+    echo "${pipeline_version_str}" > pipeline_version.txt
+
+    # When WTS data is present, ORANGE expects the somatic SAGE VCF to have appended WTS data; CS indicates this should
+    # occur after PURPLE. Since ORANGE only collects the somatic SAGE VCF from the PURPLE output directory, we must
+    # prepare accordingly
+
+    # Isofox inputs are also expected to have the tumor sample ID in the filename
+
+    # NOTES(SW): Use of symlinks was causing reliability issues on HPC with Singularity, switched to full file copy instead
```

Comment:
Do you remove the copied files afterwards? How big are they?

Reply:
The copies are not removed and are negligible in size relative to other inputs. You can expect the typical output size being:

- PURPLE: <20 MB
- SAGE append: <5 MB
- Isofox: <100 MB

This is against the background of MarkDups tumor output usually being >100,000 MB so the extra disk usage overhead of ~125 MB in ORANGE won't be a concern to users